policy/modules/apps/cdrecord.if

```diff
@@ -49,3 +49,32 @@ interface(`cdrecord_exec',`
 	corecmd_search_bin($1)
 	can_exec($1, cdrecord_exec_t)
 ')
+
+########################################
+## <summary>
+##	Role access for cdripper.
+## </summary>
+## <param name="role">
+##	<summary>
+##	Role allowed access.
+##	</summary>
+## </param>
+## <param name="domain">
+##	<summary>
+##	User domain for the role.
+##	</summary>
+## </param>
+#
+interface(`cdripper_role',`
+	gen_require(`
+		attribute_role cdripper_roles;
+		type cdripper_t, cdripper_exec_t;
+	')
+
+	roleattribute $1 cdripper_roles;
+
+	domtrans_pattern($2, cdripper_exec_t, cdripper_t)
+
+	allow $2 cdripper_t:process { ptrace signal_perms };
+	ps_process_pattern($2, cdripper_t)
+')
```

Comment:
I see no reason for a `cdripper_role()`. You should probably use a `cdrecord_domtrans_cdripper()` and `cdrecord_run_cdripper()` instead

Reply:
The two roles are independent of each other, in other words cdrecord_role() does not imply cdripper_role() and viceversa.

People might want to enable one but not the other.

So this is a nack.

Reply:
Why did you decide to append this cdripper domain to the existing cdrecord module?

These are two fundamentally different things. CD record writes to the removable device and reads from your home dir (example: burning ~/fedora.iso to a CD) CD ripper does the opposite. It reads the CD and writes it to your home directory.

Reply:
Because they both belong to the same package which is cdrtools, so it doesn't make sense to create to different modules for programs that belong to the same package !

Reply:
Ok, but i think you might have misunderstood the point i was trying to make.

When i say that you should use a `cdrecord_run_cdripper()` instead of `cdripper_role()`, i am not referring to the type cdrecord_t, but instead i am referring to the cdrecord module. the cdripper_t type is declared in the cdrecord module and so when you create new interfaces for cdripper, you should prepend the name with cdrecord_ to indicate what module the interface is part of.

So even if you would for some reason stick with `cdripper_role()` you would (probably) still need to rename it to `cdrecord_cdripper_role()`.

Howver there is no need for a "role" here. And you should not argue that you need a role for cdripper because cdrecord uses one. cdripper is not cdrecord and vice versa.

Reply:
The name is definitely wrong, as @doverride noted.